lib/matplotlib/axes/_base.py

```diff
@@ -3933,25 +3934,26 @@ def twiny(self):
         """
         Create a twin Axes sharing the yaxis
 
-        Create a twin of Axes for generating a plot with a shared
-        y-axis but independent x-axis.  The x-axis of self will have
-        ticks on bottom and the returned axes will have ticks on the
-        top.
+        Create a new Axes instance with an invisible y-axis and an independent
+        x-axis positioned opposite to the original one (i.e. at top). The
+        y-axis autoscale setting will be inherited from the original Axes.
+        To ensure that the tick marks of both x-axes align, see
+        `~matplotlib.ticker.LinearLocator`
```

Comment:
This is the original wording - do you have any suggestions for a better sentence? (or you could incorporate it in your PR since that deals with tickers)